external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/CachedKafkaConsumer.scala

```diff
@@ -147,20 +155,14 @@ object CachedKafkaConsumer extends Logging {
       groupId: String,
       topic: String,
       partition: Int,
-      kafkaParams: ju.Map[String, Object]): CachedKafkaConsumer[K, V] =
-    CachedKafkaConsumer.synchronized {
+      kafkaParams: ju.Map[String, Object]): CachedKafkaConsumer[K, V] = {
       val k = CacheKey(groupId, topic, partition)
-      val v = cache.get(k)
-      if (null == v) {
-        logInfo(s"Cache miss for $k")
-        logDebug(cache.keySet.toString)
-        val c = new CachedKafkaConsumer[K, V](groupId, topic, partition, kafkaParams)
-        cache.put(k, c)
-        c
-      } else {
-        // any given topicpartition should have a consistent key and value type
-        v.asInstanceOf[CachedKafkaConsumer[K, V]]
-      }
+      val v = cache.get(k, new Callable[CachedKafkaConsumer[_, _]] {
+        override def call(): CachedKafkaConsumer[K, V] = {
+          new CachedKafkaConsumer[K, V](groupId, topic, partition, kafkaParams)
```

Comment:
I think it's worth keeping the info / debug level logging for when a cache miss has happened.